dns/check-zone.sh

```diff
@@ -17,6 +31,7 @@ docker run -ti \
        -v ~/.config/gcloud:/.config/gcloud:ro \
        -v `pwd`/octodns-config.yaml:/octodns/config.yaml:ro \
        -v `pwd`/zone-configs:/octodns/config:ro \
+       -e GOOGLE_CLOUD_PROJECT \
```

Comment:
what does this do with no value?

Reply:
Pulls the current value from the environment, which, in this case, is set in the Makefile.

Reply:
Fascinating.  I did not know this.

Reply:
Why do we need this?  The project is in the config...

Reply:
was getting this:
```
2019-04-27T00:35:47  [140458153098088] WARNING google.auth._default No project ID could be determined. Consider running `gcloud config set project` or setting the GOOGLE_CLOUD_PROJECT environment variable
```

Reply:
Mine passes even without it set -- it doesn't actually need that.  Does yours fail?

```
$ ./check-zone.sh k8s.io.
Checking that the GCP dns servers for k8s.io. serve up everything in our octodns config
2019-04-29T22:01:20  [139806960397192] INFO  Manager __init__: config_file=/octodns/config.yaml
2019-04-29T22:01:20  [139806960397192] INFO  Manager __init__:   max_workers=1
2019-04-29T22:01:20  [139806960397192] INFO  Manager __init__:   max_workers=False
/usr/local/lib/python2.7/site-packages/google/auth/_default.py:66: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a "quota exceeded" or "API not enabled" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/.
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
2019-04-29T22:01:20  [139806960397192] WARNING google.auth._default No project ID could be determined. Consider running `gcloud config set project` or setting the GOOGLE_CLOUD_PROJECT environment variable
2019-04-29T22:01:20  [139806960397192] INFO  GoogleCloudProvider[gcp] populate: found 56 records, exists=True
Using GCP project kubernetes-public
name,type,ttl,ns-cloud-d1.googledomains.com.,ns-cloud-d2.googledomains.com.,ns-cloud-d3.googledomains.com.,ns-cloud-d4.googledomains.com.,consistent
2019-04-29T22:01:20  [139806960397192] INFO  check-zone server=216.239.32.109
2019-04-29T22:01:20  [139806960397192] INFO  check-zone server=216.239.34.109
2019-04-29T22:01:20  [139806960397192] INFO  check-zone server=216.239.36.109
2019-04-29T22:01:20  [139806960397192] INFO  check-zone server=216.239.38.109
*.docs.k8s.io.,CNAME,300,netlifyglobalcdn.com.,netlifyglobalcdn.com.,netlifyglobalcdn.com.,netlifyglobalcdn.com.,True
7ujw4gp3z2cq.testgrid.k8s.io.,CNAME,300,gv-t2drtmd73nsutd.dv.googlehosted.com.,gv-t2drtmd73nsutd.dv.googlehosted.com.,gv-t2drtmd73nsutd.dv.googlehosted.com.,gv-t2drtmd73nsutd.dv.googlehosted.com.,True
k8s.io.,A,300,35.201.71.162,35.201.71.162,35.201.71.162,35.201.71.162,True
k8s.io.,MX,300,1 aspmx.l.google.com. 5 alt1.aspmx.l.google.com. 5 alt2.aspmx.l.google.com. 10 alt3.aspmx.l.google.com. 10 alt4.aspmx.l.google.com.,1 aspmx.l.google.com. 5 alt1.aspmx.l.google.com. 5 alt2.aspmx.l.google.com. 10 alt3.aspmx.l.google.com. 10 alt4.aspmx.l.google.com.,1 aspmx.l.google.com. 5 alt1.aspmx.l.google.com. 5 alt2.aspmx.l.google.com. 10 alt3.aspmx.l.google.com. 10 alt4.aspmx.l.google.com.,1 aspmx.l.google.com. 5 alt1.aspmx.l.google.com. 5 alt2.aspmx.l.google.com. 10 alt3.aspmx.l.google.com. 10 alt4.aspmx.l.google.com.,True
k8s.io.,NS,21600,ns-cloud-d1.googledomains.com. ns-cloud-d2.googledomains.com. ns-cloud-d3.googledomains.com. ns-cloud-d4.googledomains.com.,ns-cloud-d1.googledomains.com. ns-cloud-d2.googledomains.com. ns-cloud-d3.googledomains.com. ns-cloud-d4.googledomains.com.,ns-cloud-d1.googledomains.com. ns-cloud-d2.googledomains.com. ns-cloud-d3.googledomains.com. ns-cloud-d4.googledomains.com.,ns-cloud-d1.googledomains.com. ns-cloud-d2.googledomains.com. ns-cloud-d3.googledomains.com. ns-cloud-d4.googledomains.com.,True
k8s.io.,TXT,300,v=spf1 include:_spf.google.com ~all google-site-verification=RJbZ_ganmSWvslSKOBG-QHv62XTjJZcigpWIFttStFs,v=spf1 include:_spf.google.com ~all google-site-verification=RJbZ_ganmSWvslSKOBG-QHv62XTjJZcigpWIFttStFs,v=spf1 include:_spf.google.com ~all google-site-verification=RJbZ_ganmSWvslSKOBG-QHv62XTjJZcigpWIFttStFs,v=spf1 include:_spf.google.com ~all google-site-verification=RJbZ_ganmSWvslSKOBG-QHv62XTjJZcigpWIFttStFs,True
_acme-challenge.docs.k8s.io.,A,300,0.0.0.0,0.0.0.0,0.0.0.0,0.0.0.0,True
_github-challenge-kubernetes-client.k8s.io.,TXT,300,8135eb1976,8135eb1976,8135eb1976,8135eb1976,True
_github-challenge-kubernetes-csi.k8s.io.,TXT,300,5aa2511ab7,5aa2511ab7,5aa2511ab7,5aa2511ab7,True
_github-challenge-kubernetes-incubator.k8s.io.,TXT,300,cc457b8e52,cc457b8e52,cc457b8e52,cc457b8e52,True
_github-challenge-kubernetes-retired.k8s.io.,TXT,300,b36f0d3bca,b36f0d3bca,b36f0d3bca,b36f0d3bca,True
_github-challenge-kubernetes-sigs.k8s.io.,TXT,300,c576f646fc,c576f646fc,c576f646fc,c576f646fc,True
_github-challenge-kubernetes.k8s.io.,TXT,300,8d02d39186,8d02d39186,8d02d39186,8d02d39186,True
apt.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
aupoulkmindh.gubernator.k8s.io.,CNAME,300,gv-afluggfkiwfho7.dv.googlehosted.com.,gv-afluggfkiwfho7.dv.googlehosted.com.,gv-afluggfkiwfho7.dv.googlehosted.com.,gv-afluggfkiwfho7.dv.googlehosted.com.,True
blog.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
2019-04-29T22:01:21  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: canary.k8s.io.
2019-04-29T22:01:21  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: canary.k8s.io.
2019-04-29T22:01:21  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: canary.k8s.io.
2019-04-29T22:01:21  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: canary.k8s.io.
canary.k8s.io.,NS,300,,,,,True
changelog.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
ci-test.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
code.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
cs.k8s.io.,A,300,147.75.97.58,147.75.97.58,147.75.97.58,147.75.97.58,True
devstats.k8s.io.,CNAME,300,k8s.devstats.cncf.io.,k8s.devstats.cncf.io.,k8s.devstats.cncf.io.,k8s.devstats.cncf.io.,True
discuss.k8s.io.,CNAME,300,kubernetes.hosted-by-discourse.com.,kubernetes.hosted-by-discourse.com.,kubernetes.hosted-by-discourse.com.,kubernetes.hosted-by-discourse.com.,True
dl.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
docs.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
examples.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
feature.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
features.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
gcsweb.k8s.io.,A,300,35.244.175.21,35.244.175.21,35.244.175.21,35.244.175.21,True
get.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
git.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
go.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
gubernator.k8s.io.,A,300,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,True
gubernator.k8s.io.,AAAA,300,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,True
issue.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
issues.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
kind.sigs.k8s.io.,CNAME,300,k8s-kind.netlify.com.,k8s-kind.netlify.com.,k8s-kind.netlify.com.,k8s-kind.netlify.com.,True
node-perf-dash.k8s.io.,A,300,130.211.155.47,130.211.155.47,130.211.155.47,130.211.155.47,True
perf-dash.k8s.io.,A,300,35.188.102.189,35.188.102.189,35.188.102.189,35.188.102.189,True
pr-test.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
pr.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
prow.k8s.io.,A,300,35.186.196.185,35.186.196.185,35.186.196.185,35.186.196.185,True
prs.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
redirect.k8s.io.,A,300,35.201.71.162,35.201.71.162,35.201.71.162,35.201.71.162,True
rel.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
releases.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
sigs.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
slack.k8s.io.,A,300,130.211.42.98,130.211.42.98,130.211.42.98,130.211.42.98,True
spartakus.k8s.io.,A,300,107.178.240.20,107.178.240.20,107.178.240.20,107.178.240.20,True
submit-queue.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
2019-04-29T22:01:23  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: test-cncf-aws.k8s.io.
2019-04-29T22:01:23  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: test-cncf-aws.k8s.io.
2019-04-29T22:01:23  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: test-cncf-aws.k8s.io.
2019-04-29T22:01:23  [139806960397192] INFO  check-zone *** NS Record with NoAnswer for: test-cncf-aws.k8s.io.
test-cncf-aws.k8s.io.,NS,300,,,,,True
testgrid.k8s.io.,A,300,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,216.239.32.21 216.239.34.21 216.239.36.21 216.239.38.21,True
testgrid.k8s.io.,AAAA,300,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,2001:4860:4802:32::15 2001:4860:4802:34::15 2001:4860:4802:36::15 2001:4860:4802:38::15,True
velodrome.k8s.io.,A,300,104.197.200.129,104.197.200.129,104.197.200.129,104.197.200.129,True
www.k8s.io.,CNAME,300,k8s.io.,k8s.io.,k8s.io.,k8s.io.,True
yum.k8s.io.,CNAME,300,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,redirect.k8s.io.,True
***PASS***
```